core/templates/pages/story-viewer-page/navbar-breadcrumb/story-viewer-navbar-breadcrumb.component.ts

```diff
@@ -13,47 +13,63 @@
 // limitations under the License.
 
 /**
- * @fileoverview Directive for the navbar breadcrumb of the story viewer.
+ * @fileoverview Component for the navbar breadcrumb of the story viewer.
  */
 
-require('domain/classroom/classroom-domain.constants.ajs.ts');
-require('domain/story_viewer/story-viewer-backend-api.service.ts');
-require('domain/utilities/url-interpolation.service.ts');
-require('services/contextual/url.service.ts');
-
+import { ClassroomDomainConstants } from 'domain/classroom/classroom-domain.constants';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { StoryViewerBackendApiService } from 'domain/story_viewer/story-viewer-backend-api.service';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
 import { Subscription } from 'rxjs';
+import { UrlService } from 'services/contextual/url.service';
+
+@Component({
+  selector: 'oppia-story-viewer-navbar-breadcrumb',
+  templateUrl: './story-viewer-navbar-breadcrumb.component.html',
+  styleUrls: []
+})
+export class StoryViewerNavbarBreadcrumbComponent implements OnInit, OnDestroy {
+  topicName: string;
+  storyTitle: string;
+  topicUrlFragment: string;
+  classroomUrlFragment: string;
+  storyUrlFragment: string;
+  constructor(
+    private storyViewerBackendApiService: StoryViewerBackendApiService,
+    private urlInterpolationService: UrlInterpolationService,
+    private urlService: UrlService
+  ) {}
+
+  directiveSubscriptions = new Subscription();
+  getTopicUrl(): string {
+    return this.urlInterpolationService.interpolateUrl(
+      ClassroomDomainConstants.TOPIC_VIEWER_STORY_URL_TEMPLATE, {
+        topic_url_fragment: this.topicUrlFragment,
+        classroom_url_fragment: this.classroomUrlFragment,
+        story_url_fragment: this.storyUrlFragment
+      });
+  }
 
-angular.module('oppia').component('storyViewerNavbarBreadcrumb', {
-  template: require('./story-viewer-navbar-breadcrumb.component.html'),
-  controller: [
-    'StoryViewerBackendApiService', 'UrlInterpolationService',
-    'UrlService', 'TOPIC_VIEWER_STORY_URL_TEMPLATE',
-    function(
-        StoryViewerBackendApiService, UrlInterpolationService,
-        UrlService, TOPIC_VIEWER_STORY_URL_TEMPLATE) {
-      var ctrl = this;
-      ctrl.directiveSubscriptions = new Subscription();
-      ctrl.getTopicUrl = function() {
-        return UrlInterpolationService.interpolateUrl(
-          TOPIC_VIEWER_STORY_URL_TEMPLATE, {
-            topic_url_fragment: (
-              UrlService.getTopicUrlFragmentFromLearnerUrl()),
-            classroom_url_fragment: (
-              UrlService.getClassroomUrlFragmentFromLearnerUrl())
-          });
-      };
+  ngOnInit(): void {
+    this.topicUrlFragment = this.urlService.getTopicUrlFragmentFromLearnerUrl();
+    this.classroomUrlFragment =
+     this.urlService.getClassroomUrlFragmentFromLearnerUrl();
```

Comment:
For all 2 line statements, we generally prefer doing the following I think

this.xyz  = (
    abc);

Reply:
done